src/main/flight/gyroanalyse.c

```diff
@@ -287,44 +331,39 @@ static FAST_CODE_NOINLINE void gyroDataAnalyseUpdate(gyroAnalyseState_t *state,
-                if (state->fftData[i] > state->fftData[i + 1]) {
-                    cubedData = state->fftData[i] * state->fftData[i] * state->fftData[i];
-                    fftSum += cubedData;
-                    fftWeightedSum += cubedData * (i + 1);
-                } else {
-                break;
-                }
+            float squaredData = state->fftData[binMax] * state->fftData[binMax];
+            float fftSum = squaredData;
+            float fftWeightedSum = squaredData * binMax;
+            // accumulate upper shoulder unless it would be FFT_BIN_COUNT
+            binMax += 1;
+            if (binMax < FFT_BIN_COUNT){
+                squaredData = state->fftData[binMax] * state->fftData[binMax];
+                fftSum += squaredData;
+                fftWeightedSum += squaredData * binMax;
             }
-            // accumulate lower shoulder
-            for (int i = binMax; i > binStart + 1; i--) {
-                if (state->fftData[i] > state->fftData[i - 1]) {
-                    cubedData = state->fftData[i] * state->fftData[i] * state->fftData[i];
-                    fftSum += cubedData;
-                    fftWeightedSum += cubedData * (i + 1);
-                } else {
-                break;
-                }
+            binMax -= 1;
+            // accumulate lower shoulder unless below binStart
+            if (binMax > binStart){
+                binMax -= 1;
+                squaredData = state->fftData[binMax] * state->fftData[binMax];
+                fftSum += squaredData;
+                fftWeightedSum += squaredData * binMax;
             }
-            // get weighted center of relevant frequency range (this way we have a better resolution than 31.25Hz)
+
+            // get centerFreq in Hz from weighted bins
             float centerFreq = dynNotchMaxCtrHz;
             float fftMeanIndex = 0;
-             // idx was shifted by 1 to start at 1, not 0
             if (fftSum > 0) {
-                fftMeanIndex = (fftWeightedSum / fftSum) - 1;
-                // the index points at the center frequency of each bin so index 0 is actually 16.125Hz
+                fftMeanIndex = (fftWeightedSum / fftSum);
                 centerFreq = fftMeanIndex * fftResolution;
             } else {
-                centerFreq = state->prevCenterFreq[state->updateAxis];
+                centerFreq = state->centerFreq[state->updateAxis];
             }
             centerFreq = fmax(centerFreq, dynNotchMinHz);
-            centerFreq = biquadFilterApply(&state->detectedFrequencyFilter[state->updateAxis], centerFreq);
-            state->prevCenterFreq[state->updateAxis] = state->centerFreq[state->updateAxis];
-            state->centerFreq[state->updateAxis] = centerFreq;
+
+            // PT1 style dynamic smoothing moves rapidly towards big peaks and slowly away, up to 5x faster
+            float dynamicFactor = constrainf(dataMax / dataMin, 1.0f, 5.0f);
```

Comment:
@ctzsnooze have you considered making the constraints on dynamicFactor (or perhaps just the high end constraint) tunable, either directly or with a meta-value(s) that impact the range?

I acknowledge that there are many good reasons to keep down the number of tunables, but even beyond different quads  / flight styles, things like the other filter settings (rpm vs not!) or gyro sample rate could make a big difference on the 'personality' of the gyro signal even on the same quad.  I suspect there *could* be benefit to be had from tuning the dynamicFactor range for different quads/filtering scenarios/etc.

Perhaps make it tunable for testing purposes initially (or a special build that does so) to make it accessible for testers to try different values and help determine whether it makes sense as a tunable or if there is a specific range that works best across a wide array of quads / setups better than 1.0 - 5.0.